lib/xlat_tables_v2/xlat_tables_core.c

```diff
@@ -563,6 +566,10 @@ static uintptr_t xlat_tables_map_region(xlat_ctx_t *ctx, mmap_region_t *mm,
 			end_va = xlat_tables_map_region(ctx, mm, table_idx_va,
 					       subtable, XLAT_TABLE_ENTRIES,
 					       level + 1U);
+#if !HW_ASSISTED_COHERENCY
+			flush_dcache_range((uintptr_t)subtable,
+				XLAT_TABLE_ENTRIES * sizeof(uint64_t));
```

Comment:
Why are these calls added after each call to `xlat_tables_map_region()` rather than in that function?

Reply:
There are two options:

1 - Clean the whole table after (which will only affect the modified cache lines anyway).
2 - Clean every entry as they are modified, which will increase the code complexity and clean the same cache lines multiple times.

For mapping/unmapping I've chosen 1 because we only need to do the change once. For  ``get_mem_attributes`` and ``change_mem_attributes`` I've chosen 2 because we need to clean once when invalidating the table entry and once when it is set to the new attributes.